File: DSAlgorithms/Leetcode/c++/binary_search/69-sqrt.cpp
class Solution {
public:
    int mySqrt(int x) {
       int left = 0;
       int right = x;
       int result = 0;
       while(left <= right) 
       {
           int mid = left + (right - left) / 2;
           if(((double)mid * mid) > x)
                right = mid - 1;
            else if((double)(mid * mid) < x)
            {
                left = mid + 1;
                result = mid;
            }
            else
                return mid;
       }
       return result;
    }
};

File: DSAlgorithms/Leetcode/ts/src/binary_search/69-sqrt.ts
const mySqrt = (x: number): number =>{
    let left = 0;
    let right = x;
    let result = 0;
    while(left <= right)
    {
        let mid = Math.floor(left + (right - left) /2);
        if((mid * mid) > x)
            right = mid - 1;
        else if((mid * mid) < x)
        {
            left = mid + 1;
            result = mid;
        }
        else
            return mid;
    }
    return result;
};